### PR TITLE
Replace deprecated WebRequest with HttpClient

### PR DESCRIPTION
**Pull request by AZero13**
Use HttpClient instead

### CODE REVIEW

File: src/libraries/System.Windows.Extensions/src/System/Media/SoundPlayer.cs [hunk 1]
```diff
@@ -258,16 +259,14 @@ private void LoadSync()
...
-                webResponse = webRequest.GetResponse();
+                using (HttpClient httpClient = new HttpClient())
+                {
+                    httpClient.Timeout = LoadTimeout;
 
-                // now get the stream
-                _stream = webResponse.GetResponseStream();
+                    HttpRequestMessage message = new HttpRequestMessage(HttpMethod.Post, _uri);
+                    var response = Task.Run(client.SendAsync(message, HttpCompletionOption.ResponseHeadersRead)).Result;
+                    _stream = Task.Run(response.Content.ReadAsStreamAsync()).Result;
```
Comment (teo-tsirpanis) [inline src/libraries/System.Windows.Extensions/src/System/Media/SoundPlayer.cs:268]:
```suggestion
                    var response = client.Send(message, HttpCompletionOption.ResponseHeadersRead);
                    _stream = response.Content.ReadAsStream();
```

File: src/libraries/System.Windows.Extensions/src/System/Media/SoundPlayer.cs [hunk 2]
```diff
@@ -483,13 +482,12 @@ private async Task CopyStreamAsync(CancellationToken cancellationToken)
...
-                    WebRequest webRequest = WebRequest.Create(_uri);
-#pragma warning restore SYSLIB0014
-                    using (cancellationToken.Register(r => ((WebRequest)r!).Abort(), webRequest))
+
+                    using (HttpClient httpClient = new HttpClient())
                     {
-                        WebResponse webResponse = await webRequest.GetResponseAsync().ConfigureAwait(false);
-                        _stream = webResponse.GetResponseStream();
+                        HttpRequestMessage message = new HttpRequestMessage(HttpMethod.Post, _uri);
+                        var response = await client.SendAsync(message, HttpCompletionOption.ResponseHeadersRead);
```
Comment (teo-tsirpanis) [inline src/libraries/System.Windows.Extensions/src/System/Media/SoundPlayer.cs:489]:
```suggestion
                        var response = await client.SendAsync(message, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
```

**Review by teo-tsirpanis — CHANGES_REQUESTED** [review]

**Comment by hez2010** [comment]
I don't think HttpClient should be used as transient, instead an IHttpClientFactory should be used to avoid socket exhaustion.

**Comment by AtariDreams** [comment]
How can I do this when I need to have default credentials

**Comment by AtariDreams** [comment]
@stephentoub Is this an appropriate use of HttpClient and its properties/methods?

**Comment by jkotas** [comment]
HttpClient handles http: and https: protocols only. WebRequest handles file: and ftp: protocols as well. So this change will break callers that pass urls that use these additional protocols. I would be concerned about file: in particular.

The libraries that you are changing are provided for compatibility with .NET Framework. We are not investing in them and it is fine for them to use use WebRequest in them since that is only provided for compatibility with .NET Framework. We plan to document which libraries are actively maintained vs. libraries are in sustained engineering mode to create clarity about this. cc @jeffhandley

**Comment by stephentoub** [comment]
@AtariDreams, thanks for working on this.  I agree with Jan; while in theory it'd be nice to get rid of the use of WebRequest, replacing it with HttpClient is a breaking change, and these libraries are primarily in maintenance mode.  Given that, I'm going to close this, and also https://github.com/dotnet/runtime/issues/41516, which may have steered you wrong (apologies).  Thanks for your interest in contributing.

**Pull request closed without merging**